spec.html

```diff
@@ -1223,3 +1223,192 @@ <h1>Execution Contexts</h1>
 </emu-clause>
 </emu-clause>
 
+  <!-- es6num="11.9" -->
+  <emu-clause id="sec-automatic-semicolon-insertion">
+    <h1>Automatic Semicolon Insertion</h1>
+    <p>Most ECMAScript statements and declarations <del>must be</del><ins>are logically</ins> terminated with a semicolon. Such semicolons may always appear explicitly in the source text. For convenience, however, such semicolons may be omitted from the source text in certain situations. These situations are described by saying that semicolons are automatically inserted into the source code token stream in those situations.</p>
```

Comment:
Can you elaborate on what you hope to convey with "must be" → "are logically"?

I worry that appearing to lighten this language will be used as ammo in future ASI debates in the community.

Reply:
I'm fine to drop this; the change in text here is actually irrelevant, and I just did it because I was trying to tighten up the whole ASI section. I'd be fine to back it out.

The word "must" here is a bit confusing. It's a strange sense of must, because clearly you *can* leave out the semicolon. I believe what the sentence is saying is that there is a logical semicolon at the end of these statements, rather than necessarily a textual one. This sentence is non-normative and the beginning of some expository text describing what ASI is at a high level, so I don't know if it's the right time to be super-technical, where it's trying to say, roughly, "After ASI runs, most statements must end in a semicolon".

Reply:
The slightly facetious way I usually describe it is "semicolons are SO required that the engine will try to put them in for you if you fail to include them explicitly"; probably too harsh for even a non-normative part of the spec, of course.

Reply:
That wording also seems good to me, and more straightforward than the current wording. Do you have another suggestion that would fit into how the rest of the spec is worded? Looks like the wording goes all the way back to ES1.

cc @allenwb @bterlson